### ISSUE TITLE
Truffle 4 (beta) fails to find an artefact for a contract, once renamed

### ISSUE THREAD

**Issue by elenadimitrova** [trufflesuite/truffle#580]
## Issue

The Json artefact for a contract, once renamed cannot be found.
`Error: Could not find artifacts for ./MyContract.sol from any sources`

## Steps to Reproduce

Run `truffle compile`
Rename the resulting json e.g. `MyContract_4.json`
Run `truffle test`

## Expected Behavior
Tests using the MyContract artefact to succeed.

## Actual Results

```
/node_modules/truffle/build/cli.bundled.js:58898
  throw new Error("Could not find artifacts for " + import_path + " from any sources");
  ^

Error: Could not find artifacts for ./MyContract.sol from any sources
    at Resolver.require (/node_modules/truffle/build/cli.bundled.js:58898:9)
    at TestResolver.require (/node_modules/truffle/build/cli.bundled.js:200045:30)
    at Object.require (/node_modules/truffle/build/cli.bundled.js:86271:36)
    at ResolverIntercept.require (/node_modules/truffle/build/cli.bundled.js:336787:32)
    at /migrations/4_setup_script.js:2:26
    at ContextifyScript.Script.runInContext (vm.js:35:29)
    at ContextifyScript.Script.runInNewContext (vm.js:41:15)
    at /node_modules/truffle/build/cli.bundled.js:200794:14
    at tryToString (fs.js:456:3)
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:443:12)
1
Cleaning up. Killing child process 99566...
```

## Environment

* Truffle version: 4 Beta
* Ethereum client: testrpc 4.1.1
* node version: 6.10.3
* npm version: 5.4.2

**Comment by tcoulter** [comment]
Hi Elena, 

This is likely due to your `artifacts.require()` statement. I'm guessing your `artifacts.require()` statement looks like this:

```
var MyContract = artifacts.require("MyContract.sol");
```

If so, you can get around this just by referring to the artifact file by name:

```
var MyContract = artifacts.require("MyContract_4");
```

Note that there's no need for the `.sol` extension.

Let me know if that works!

**Comment by pau1m** [comment]
Did you also rename the code inside the contracts to be the same as the filename?

**Comment by elenadimitrova** [comment]
My contract is called `MyContract` but the filename its contained in changes over time (to provide different versions of the same contract to the front-end).
Therefore I'd not expect to have to change the `artifacts.require` statement if the filename changes.
Also this was not an issue with truffle 3

**Comment by rakibgalib** [comment]
try to recheck your contract name.if your contract name doesn't match with your file name.Ex.File name:myContract.sol and contact name within the file need to be "Contact myContact{}". Hopefully it will help you

**Comment by cgewecke** [comment]
Hi @elenadimitrova Apologies for closing this early, it looks like we accidentally used the word 'fixed' in a merge commit. @gnidan is optimistic that a recent PR has resolved this - it should be available on 4.0.7. Please re-open if it keeps happening and thanks for reporting!
